### PR TITLE
LPS-143238 Multiple com.liferay.search.experiences.rest.resource.v1_0.test integration test failures

### PR DESCRIPTION
**Pull request by gustavolimav**
https://issues.liferay.com/browse/LPS-143238

### CODE REVIEW

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:search

File: modules/dxp/apps/search-experiences/search-experiences-rest-impl/src/main/java/com/liferay/search/experiences/rest/internal/resource/v1_0/SXPBlueprintResourceImpl.java [hunk 1]
```diff
@@ -296,6 +285,24 @@ private String _getSchemaVersion() {
 			contextUriInfo.getPath(), "v", StringPool.SLASH);
 	}
 
+	private void _setActions(long sxpBlueprintId, SXPBlueprint sxpBlueprint) {
```
Comment (peerkar) [inline modules/dxp/apps/search-experiences/search-experiences-rest-impl/src/main/java/com/liferay/search/experiences/rest/internal/resource/v1_0/SXPBlueprintResourceImpl.java:288]: @gustavolimav no need to pass the ID as it's already within the SXPBlueprint. Also if you change this method to return the SXPBlueprint, you can simplify the related _setAction() calls.

File: modules/dxp/apps/search-experiences/search-experiences-rest-impl/src/main/java/com/liferay/search/experiences/rest/internal/resource/v1_0/SXPBlueprintResourceImpl.java [hunk 2]
```diff
@@ -226,7 +211,7 @@ public SXPBlueprint postSXPBlueprint(SXPBlueprint sxpBlueprint)
 
 		SXPBlueprintUtil.unpack(sxpBlueprint);
 
-		return _sxpBlueprintDTOConverter.toDTO(
+		SXPBlueprint sxpBlueprint1 = _sxpBlueprintDTOConverter.toDTO(
```
Comment (peerkar) [inline modules/dxp/apps/search-experiences/search-experiences-rest-impl/src/main/java/com/liferay/search/experiences/rest/internal/resource/v1_0/SXPBlueprintResourceImpl.java:214]: @gustavolimav could be simplified as:

```
		return _setActions(_sxpBlueprintDTOConverter.toDTO(
			new DefaultDTOConverterContext(
				contextAcceptLanguage.isAcceptAllLanguages(), new HashMap<>(),
				_dtoConverterRegistry, contextHttpServletRequest,
				sxpBlueprint.getId(),
				contextAcceptLanguage.getPreferredLocale(), contextUriInfo,
				contextUser),
			_sxpBlueprintService.addSXPBlueprint(
				_getConfigurationJSON(sxpBlueprint),
				LocalizedMapUtil.getLocalizedMap(
					contextAcceptLanguage.getPreferredLocale(),
					sxpBlueprint.getDescription(),
					sxpBlueprint.getDescription_i18n()),
				_getElementInstancesJSON(sxpBlueprint), _getSchemaVersion(),
				LocalizedMapUtil.getLocalizedMap(
					contextAcceptLanguage.getPreferredLocale(),
					sxpBlueprint.getTitle(), sxpBlueprint.getTitle_i18n()),
				ServiceContextFactory.getInstance(contextHttpServletRequest))));
```

File: modules/dxp/apps/search-experiences/search-experiences-rest-impl/src/main/java/com/liferay/search/experiences/rest/internal/resource/v1_0/SXPBlueprintResourceImpl.java [hunk 3]
```diff
@@ -172,22 +172,7 @@ public Response getSXPBlueprintExport(Long sxpBlueprintId)
...
-						() -> addAction(
-							ActionKeys.DELETE, "deleteSXPBlueprint",
-							permissionName, sxpBlueprintId)
-					).put(
-						"view",
-						() -> addAction(
-							ActionKeys.VIEW, "getSXPBlueprint", permissionName,
-							sxpBlueprintId)
-					).build());
+				_setActions(sxpBlueprintId, sxpBlueprint);
```
Comment (peerkar) [inline modules/dxp/apps/search-experiences/search-experiences-rest-impl/src/main/java/com/liferay/search/experiences/rest/internal/resource/v1_0/SXPBlueprintResourceImpl.java:175]: Same here: return _setActions(...

**Review by peerkar — COMMENTED** [review]
@gustavolimav some minor simplifications could be made. Same to  SXPElementResource as commented to SXPBlueprintResource. Otherwise, looks good to me.

@BryanEngler can you please confirm whether those failures are flakes only.

**Comment by gustavolimav** [comment]
ci:test:search

**Comment by gustavolimav** [comment]
ci:test:search

**Comment by gustavolimav** [comment]
ci:sf

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:search

**Comment by BryanEngler** [comment]
ci:test:relevant

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:search

**Pull request closed without merging**